docs/variant.md

```diff
@@ -1,207 +1,200 @@
 
 ```reason
-/* Zoo.re */
-
-type animal = Dog | Cat | Bird;
+let fluffy = Cat;
+let spot = Dog;
 ```
 
-```reason
-/* example.re */
+## Constructor Arguments
 
-let pet: Zoo.animal = Dog; /* preferred */
-/* or */
-let pet = Zoo.Dog;
-```
-
-### Constructor Arguments
-
-A variant's constructors can hold extra data separated by comma.
+Variant constructors can have data associated with them:
 
 ```reason
-type account =
-  | None
-  | Instagram(string)
-  | Facebook(string, int);
+type linkedList =
+  | Node(int, linkedList)
+  | Empty;
 ```
 
-Here, `Instagram` holds a `string`, and `Facebook` holds a `string` and an `int`. Usage:
+The constructors are now like functions that accept their data as arguments.
+Creating a linked list `[1, 2, 3]` looks like:
 
 ```reason
-let myAccount = Facebook("Josh", 26);
-let friendAccount = Instagram("Jenny");
+let x = Node(1, Node(2, Node(3, Empty)));
 ```
 
-**Notice how using a constructor is like calling a function**? It's as if `Facebook` was a function that accepts two arguments. This isn't a coincidence; there's a reason why a constructor's data is called "constructor argument".
+_Note: This is how the core [`list`](basic-structures.md#list) structure is
+actually implemented. The constructors are just hidden behind some syntax._
 
-Using `switch`, you can pattern-match (again, described in a later section) a constructor's arguments:
-
-```reason
-let greeting =
-  switch (myAccount) {
-  | None => "Hi!"
-  | Facebook(name, age) => "Hi " ++ name ++ ", you're " ++ string_of_int(age) ++ "-year-old."
-  | Instagram(name) => "Hello " ++ name ++ "!"
-  };
-```
-
-### Honorable Mentions
-
-The [standard library](/api/index.html) exposes two important variants you'll come to hear a lot.
+## Usage
 
-#### `option`
+The primary way to interact with variants is through
+[pattern matching](pattern-matching.md). When pattern matching over a variant
+the compiler can ensure that all cases are covered exhaustively so that there
+is no undefined behavior. This is also helpful when adding new cases to a
+variant later, the compiler will present a clear list of code that needs
+to be updated.
 
 ```reason
-type option('a) = None | Some('a);
+let interact = (animal) => {
+  switch (animal) {
+  | Cat => "pet"
+  | Dog => "throw ball"
+  | Horse => "ride"
+  | Snake => "run away"
+  };
+};
 ```
 
-This is the convention used to simulate a "nullable" (aka `undefined` or `null`) value in other languages. Thanks to this convenience type definition, Reason can default every value to be non-nullable. An `int` will always be an int, never "`int` **or** `null` **or** `undefined`". If you do want to express a "nullable int", you'd use `option(int)`, whose possible values are `None` or `Some(int)`. `switch` forces you to handle both cases; therefore, **a pure Reason program doesn't have null errors**.
-
-#### `list`
+It is possible to interact with a constructor's data while pattern matching:
```

Comment:
Maybe we can forgo this example since it is covered extensively in the pattern matching doc?

Reply:
Or do you think it makes sense to eventually merge pattern matching and variants?

Reply:
I wanted to get the bare-minimum for interacting with variants in the actual variant document so that someone reading wouldn't have to go through the pattern matching doc to find this as well. I felt that a simple switch and then a switch that uses the data were a nice medium here. I think a little duplication between the docs are okay